compiler/qsc_qasm/src/tests/declaration/array.rs

```diff
@@ -4,31 +4,425 @@
 mod bit;
 mod qubit;
 
```

Comment:
We should have a test that 8 dimensional arrays are unsupported but 7 dimensional are supported.